Build failure: datadog-agent (needs upgrade 7.56.2 -> 7.67.0)

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. `nix-shell -p datadog-agent`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/300316089

### Relevant log output

```console
/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/include/python3.13/pylifecycle.h:37:38: note: declared here
   37 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);
      |                                      ^~~~~~~~~~~~~~~~~
/build/rtloader/three/three.cpp:88:5: error: '_Py_SetProgramFullPath' was not declared in this scope; did you mean 'Py_GetProgramFullPath'?
```

### Additional context

According to https://github.com/DataDog/datadog-agent/issues/34076#issuecomment-2992715155, this should be solved by upgrading datadog-agent to 7.67.x.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.28, NixOS, 25.11 (Xantusia), 25.11.20250613.ee930f9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/xnr2xjs0clkhlhqfc3vqcmhnnwdk3c5p-source`

### Notify maintainers

@thoughtpolice @domenkozar

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/networking/dd-agent/datadog-agent.nix
+++ b/pkgs/tools/networking/dd-agent/datadog-agent.nix
@@ -18,17 +18,17 @@
 
 let
   # keep this in sync with github.com/DataDog/agent-payload dependency
-  payloadVersion = "5.0.124";
+  payloadVersion = "5.0.164";
   python = pythonPackages.python;
   owner = "DataDog";
   repo = "datadog-agent";
   goPackagePath = "github.com/${owner}/${repo}";
-  version = "7.56.2";
+  version = "7.70.2";
 
   src = fetchFromGitHub {
     inherit owner repo;
-    rev = version;
-    hash = "sha256-rU3eg92MuGs/6r7oJho2roeUCZoyfqYt1xOERoRPqmQ=";
+    tag = version;
+    hash = "sha256-yXtybHWrm+6kWW396FLlRZI0YVuThGuLfSYzoNXAEBU=";
   };
   rtloader = stdenv.mkDerivation {
     pname = "datadog-agent-rtloader";
@@ -49,11 +49,7 @@ buildGoModule rec {
 
   doCheck = false;
 
-  vendorHash =
-    if stdenv.hostPlatform.isDarwin then
-      "sha256-3Piq5DPMTZUEjqNkw5HZY25An2kATX6Jac9unQfZnZc="
-    else
-      "sha256-FR0Et3DvjJhbYUPy9mpN0QCJ7QDU4VRZFUTL0J1FSXw=";
+  vendorHash = "sha256-iWOwhfSI7mLmDy6yewV0h9Y4pjYAV6Tz6TxsINOxYMg=";
 
   subPackages = [
     "cmd/agent"
@@ -67,10 +63,14 @@ buildGoModule rec {
     makeWrapper
   ];
   buildInputs = [ rtloader ] ++ lib.optionals withSystemd [ systemd ];
-  PKG_CONFIG_PATH = "${python}/lib/pkgconfig";
+
+  proxyVendor = true;
+
+  env.PKG_CONFIG_PATH = "${python}/lib/pkgconfig";
 
   tags = [
     "ec2"
+    "kubelet"
     "python"
     "process"
     "log"
@@ -97,7 +97,7 @@ buildGoModule rec {
   postPatch = ''
     sed -e "s|PyChecksPath =.*|PyChecksPath = filepath.Join(_here, \"..\", \"${python.sitePackages}\")|" \
         -e "s|distPath =.*|distPath = filepath.Join(_here, \"..\", \"share\", \"datadog-agent\")|" \
-        -i cmd/agent/common/path/path_nix.go
+        -i pkg/util/defaultpaths/path_nix.go
     sed -e "s|/bin/hostname|${lib.getBin hostname}/bin/hostname|" \
         -i pkg/util/hostname/fqdn_nix.go
   '';

--- a/pkgs/tools/networking/dd-agent/integrations-core.nix
+++ b/pkgs/tools/networking/dd-agent/integrations-core.nix
@@ -42,14 +42,14 @@
 
 let
   inherit (lib) attrValues mapAttrs;
+  version = "7.70.2";
 
   src = fetchFromGitHub {
     owner = "DataDog";
     repo = "integrations-core";
-    rev = version;
-    sha256 = "sha256-p5eoNNHQQl314mfUk2t3qQaerPu02GKA+tKkAY7bojk=";
+    tag = version;
+    hash = "sha256-3H8nQpy/m53ZjtDfe6s89yowBXnPt+1ARfWxcx+JwQM=";
   };
-  version = "7.56.2";
 
   # Build helper to build a single datadog integration package.
   buildIntegration =
@@ -75,19 +75,13 @@ let
     pname = "checks-base";
     sourceRoot = "datadog_checks_base";
 
-    # Make setuptools build the 'base' and 'checks' modules.
-    postPatch = ''
-      substituteInPlace setup.py \
-        --replace "from setuptools import setup" "from setuptools import find_packages, setup" \
-        --replace "packages=['datadog_checks']" "packages=find_packages()"
-    '';
-
-    propagatedBuildInputs = with python3Packages; [
+    dependencies = with python3Packages; [
       binary
       cachetools
       cryptography
       immutables
       jellyfish
+      lazy-loader
       prometheus-client
       protobuf
       pydantic

